query_execution/QueryContext.cpp

```diff
@@ -58,7 +59,8 @@ QueryContext::QueryContext(const serialization::QueryContext &proto,
                            const CatalogDatabaseLite &database,
                            StorageManager *storage_manager,
                            const tmb::client_id scheduler_client_id,
-                           tmb::MessageBus *bus) {
+                           tmb::MessageBus *bus)
+    : database_(static_cast<const CatalogDatabase&>(database)) {
```

Comment:
Good catch @zuyu 

The ``database`` object is needed for the function ``getTotalTempRelationMemoryInBytes``. In this function, we need the ``CatalogRelation`` objects for all the temporary relations - to get their memory footprints. I will see if I can push this function to the ``QueryManagerSingleNode`` class.